server/routes/api/section.js

```diff
@@ -0,0 +1,159 @@
+const express = require('express');
+const router = express.Router();
+var { Course, SeatingLayout } = require('../../dbSchemas/attendanceSchema');
+
+router.get('/', (req, res) => {
+
+
+});
+
+router.get('/previousSeatingPlans', (req, res) => {
+    // this is the professor ID
+    let professor = req.body.professor;
+
+    SeatingLayout.find({$or : [{default: true} , {created_by : professor}]}, function(err, seatingLayout){
+        if(err){
+            console.log(err);
+            return res.status(500).send();
+        }
+
+        return res.status(200).json({
+            seatingLayout
+        }).send();
+    })
+
+
+});
+
+router.post('/createSeatingLayout', (req,res) => {
+    let name  = req.body.name;
+    let capacity  = req.body.capacity;
+    let dimension  = req.body.dimensions;
+    let layout  = req.body.layout;
+    let def = req.body.default;
+    let description  = req.body.description;
```

Comment:
Shouldn't there be some support so that if the createdBy and the description are not sent in the request body his doesn't fail? Cause those fields are listed as not required in the db schema

Reply:
I just tested in postman, it seems that if we don't include those fields it will create the db entry without them, and we get the 200 success code

Reply:
Which makes sense, given that they are not required. Those fields should be able to added to later in the update section endpoint if needed, right?